ledger/accountdb.go

```diff
@@ -148,25 +150,87 @@ type persistedAccountData struct {
 	round basics.Round
 }
 
-// compactAccountDeltas and accountDelta is an extension to ledgercore.AccountDeltas that is being used by the commitRound function for counting the
+//msgp:ignore persistedResourcesData
+type persistedResourcesData struct {
+	// addrid is the rowid of the account address that holds this resource.
+	addrid int64
+	// creatable index
+	aidx basics.CreatableIndex
+	// rtype is the resource type ( asset / app )
+	rtype basics.CreatableType
+	// actual resource data
+	data resourcesData
+	// the round number that is associated with the resourcesData. This field is the corresponding one to the round field
+	// in persistedAccountData, and serves the same purpose.
+	round basics.Round
+}
+
+func (prd *persistedResourcesData) AccountResource() ledgercore.AccountResource {
+	ret := ledgercore.AccountResource{
+		CreatableIndex: prd.aidx,
+		CreatableType:  prd.rtype,
+	}
+	if prd.data.IsAsset() {
+		if prd.data.IsHolding() {
+			holding := prd.data.GetAssetHolding()
+			ret.AssetHolding = &holding
```

Comment:
I had assumed ledgercore.AccountResource had pointers to allow for distinguishing nil vs empty (like the persistedResourceData does)

Reply:
It's more than that. If we were to define `AccountResource` as
```golang
type AccountResource struct {
	CreatableIndex basics.CreatableIndex
	CreatableType  basics.CreatableType

	AssetParam    basics.AssetParams
	AssetHolding  basics.AssetHolding
	AppLocalState basics.AppLocalState
	AppParams     basics.AppParams
}
```
we wouldn't be able to determine if the resource has opted-into a specific asset or not, since zero parameters of `basics.AssetParams` won't be enough.